File: 07-06-2025_Longest_span_in_an_array.java
class Solution {
    public int longestCommonSum(int[] a1, int[] a2) {
        int n=a1.length;
        HashMap<Integer,Integer> map=new HashMap<>();
        int sum=0;
        map.put(sum,-1);
        int ans=0;
        for(int i=0;i<n;i++){
            sum+=a1[i]-a2[i];
            if(map.containsKey(sum)) ans=Math.max(ans,i-map.get(sum));
            else map.put(sum,i);
        }
        return ans;
    }
}

File: 07-06-2025_Longest_span_in_an_array.cpp
class Solution {
  public:
    int longestCommonSum(vector<int> &a1, vector<int> &a2) {
        // Code here.
        int n=a1.size(),ans=0;
        unordered_map <int,int>mp;
        mp[0]=-1;
        int sum=0;
        for(int i=0;i<n;i++){
            sum+=(a1[i]-a2[i]);
            if(mp.find(sum)!=mp.end())
                ans=max(ans,i-mp[sum]);
            else
            mp[sum]=i;
        }
        return ans;
    }
};